File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        int n=strs.size();
        unordered_map<string,vector<string>> ansmap;
        for(string str:strs){
                 vector<int> store(26,0);
                 string res="";
            for(char c: str){
              store[c-'a']++;
            }
            for(int i=0;i<26;i++){
                res.append(store[i],'a'+i);
            }
            ansmap[res].push_back(str);
        }
        for(auto it:ansmap){
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> map=new HashMap<>();

        for(String str:strs){
            
            String sortedWord=str.chars().sorted().mapToObj(c->String.valueOf((char)c)).collect(Collectors.joining());


            if(!map.containsKey(sortedWord)){
                map.put(sortedWord,new ArrayList<>());
            }

            map.get(sortedWord).add(str);
        }
        return new ArrayList<>(map.values());
    }
}